pkg/route/api/validation/validation.go

```diff
@@ -109,8 +109,8 @@ func validateTLS(route *routeapi.Route) fielderrors.ValidationErrorList {
 			result = append(result, fielderrors.NewFieldInvalid("destinationCACertificate", tls.DestinationCACertificate, "edge termination does not support destination certificates"))
 		}
 	default:
-		msg := fmt.Sprintf("invalid value for termination, acceptable values are %s, %s, %s, or empty (no tls specified)", routeapi.TLSTerminationEdge, routeapi.TLSTerminationPassthrough, routeapi.TLSTerminationReencrypt)
-		result = append(result, fielderrors.NewFieldInvalid("termination", tls.Termination, msg))
+		msg := fmt.Sprintf("invalid value for termination, acceptable values are %s, %s, or %s", routeapi.TLSTerminationEdge, routeapi.TLSTerminationPassthrough, routeapi.TLSTerminationReencrypt)
+		result = append(result, fielderrors.NewFieldValueNotSupported("termination", tls.Termination, []string{msg}))
```

Comment:
rushed on this, thanks

On Wed, Nov 25, 2015 at 4:21 PM, David Eads <notifications@github.com>
wrote:

> In pkg/route/api/validation/validation.go
> <https://github.com/openshift/origin/pull/6076#discussion_r45878441>:
>
> > @@ -109,8 +109,8 @@ func validateTLS(route *routeapi.Route) fielderrors.ValidationErrorList {
> >  			result = append(result, fielderrors.NewFieldInvalid("destinationCACertificate", tls.DestinationCACertificate, "edge termination does not support destination certificates"))
> >  		}
> >  	default:
> > -		msg := fmt.Sprintf("invalid value for termination, acceptable values are %s, %s, %s, or empty (no tls specified)", routeapi.TLSTerminationEdge, routeapi.TLSTerminationPassthrough, routeapi.TLSTerminationReencrypt)
> > -		result = append(result, fielderrors.NewFieldInvalid("termination", tls.Termination, msg))
> > +		msg := fmt.Sprintf("invalid value for termination, acceptable values are %s, %s, or %s", routeapi.TLSTerminationEdge, routeapi.TLSTerminationPassthrough, routeapi.TLSTerminationReencrypt)
> > +		result = append(result, fielderrors.NewFieldValueNotSupported("termination", tls.Termination, []string{msg}))
>
> bad usage. The string slice is for valid values.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6076/files#r45878441>.
>